#!/usr/bin/env python
PACKAGE = "fm_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("K_px", double_t, 0, "The position error gain in x.", 5.0, 0.0, 20.0)
gen.add("K_py", double_t, 0, "The position error gain in y.", 5.0, 0.0, 20.0)
gen.add("K_pz", double_t, 0, "The position error gain in z.", 1.2, 0.0, 20.0)
gen.add("K_yaw", double_t, 0, "The position error gain in yaw.", 2.0, 0.0, 10.0)

gen.add("K_vx", double_t, 0, "The velocity error gain in x.", 4.0, 0.0, 20.0)
gen.add("K_vy", double_t, 0, "The velocity error gain in y.", 4.0, 0.0, 20.0)
gen.add("K_vz", double_t, 0, "The velocity error gain in z.", 0.6, 0.0, 20.0)
gen.add("K_yaw_rate", double_t, 0, "The angular velocity error gain in yaw.", 1.5, 0.0, 10.0)

gen.add("K_ax", double_t, 0, "The acceleration error gain in x.", 0.75, 0.0, 20.0)
gen.add("K_ay", double_t, 0, "The acceleration error gain in y.", 0.75, 0.0, 20.0)
gen.add("K_az", double_t, 0, "The acceleration error gain in z.", 0.0, 0.0, 20.0)
gen.add("K_yaw_acc", double_t, 0, "The angular acceleration error gain in yaw.", 0.0, 0.0, 10.0)

gen.add("v_xy_max", double_t, 0, "Maximum velocity in x-y-plane [m/s].", 8.0, 0.0, 20.0)
gen.add("v_z_up_max", double_t, 0, "Maximum velocity z-up [m/s].", 3.0, 0.0, 10.0)
gen.add("v_z_down_max", double_t, 0, "Maximum velocity z-down [m/s].", 2.0, 0.0, 10.0)
gen.add("yaw_rate_max", double_t, 0, "Maximum yaw rate [rad].", 2.0, 0.0, 6.0)

control_type_enum = gen.enum([ gen.const("Position", int_t, 0, "DJI position-yaw reference."),
                               gen.const("Velocity", int_t, 1, "DJI velocity-yaw rate reference.")],
                               "The controller type.")
gen.add("control_type", int_t, 0, "The controller type option.", 1, 0, 1, edit_method=control_type_enum)

gen.add("min_transition_altitude", double_t, 0, "Min. altitude AGL to switch from AGL to ENU.", -1.0, -1.0, 10000.0)

exit(gen.generate(PACKAGE, "Config", "PositionController"))
